File: tcp-client/tcpClient.js
const net = require("node:net")
const readline = require("readline")

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
})

const HOST = 'localhost'
const PORT = 8080

const client = new net.Socket()

client.connect(PORT, HOST, () => {
    console.log('Connected to the server')
    rl.question("Message: ", (msg) => {
        client.write(msg)
        rl.close()
    })
})

client.on('data', (data) => {
    console.log(`Received from server: ${data}`)
    client.destroy()
})

client.on('close', () => {
    console.log('Connection closed')
})

client.on('error', (error) => {
    console.error("Error:", error)
})

File: python/tcpClient.py
import socket

HOST = "127.0.0.1"
PORT = 8080

client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
client.connect((HOST, PORT))

try:
    message = input("Message: ")
    client.sendall(message.encode("utf8"))
finally:
    print("Exiting server")
    client.close()
